pretyping/nativenorm.ml

```diff
@@ -222,7 +222,12 @@ and nf_type_sort env sigma v =
   match kind_of_value v with
   | Vaccu accu -> 
       let t,s = nf_accu_type env sigma accu in
-      let s = try destSort s with DestKO -> assert false in
+      let s =
+        try
+          destSort (whd_all env s)
+        with DestKO ->
+          CErrors.anomaly (Pp.str "Value should be a sort")
+      in
```

Comment:
Why bother catching the destko?

Reply:
This was there before. I guess the idea is that `DestKO` is used as a catchable exception in the codebase. So we want to raise a fatal exception here.

Reply:
Is there a reason to expect a destko to happen here? Considering we raise an anomaly it seems strange to me.

Reply:
> Is there a reason to expect a destko to happen here?

No, except in a tricky scenario as was the case. It should certainly be a fatal error, otherwise debugging will be a pain (imagine this gets caught by unification).

I'm afraid I don't understand your concern. You feel more comfortable with raising an exception that can be silently caught by random clients?